modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -205,6 +206,18 @@ public CPDefinition addCPDefinition(
 
 		// Commerce product definition
 
+		if ((subscriptionLength < 1) || (deliverySubscriptionLength < 1)) {
+			throw new CPSubscriptionDateException(
+				"Enter subscription length greater than or equal to 1");
```

Comment:
Upon second thought, let's change the message to following format:

Subscription length must be greater than or equal to 1"

Reply:
Furthermore, I want you to create a new private method with name `isValidSubscriptionInfo` and do your condition checks there.

Reply:
@victorcdp , see my above comments for requested changes.

Reply:
Done. Should be good as well